spinnaker/resource_pipeline.go

```diff
@@ -75,13 +76,23 @@ func resourcePipelineRead(data *schema.ResourceData, meta interface{}) error {
 	pipelineName := data.Get("name").(string)
 
 	var p pipelineRead
-	if err := api.GetPipeline(client, applicationName, pipelineName, &p); err != nil {
+	jsonMap, err := api.GetPipeline(client, applicationName, pipelineName, &p)
+	if err != nil {
 		return err
 	}
 
-	err := data.Set("pipeline_id", p.Id)
+	pipeline, err := editAndEncodePipeline(jsonMap)
+	if err != nil {
+		return err
+	}
+	err = data.Set("pipeline", pipeline)
 	if err != nil {
-		return fmt.Errorf("Could not set pipeline_id for pipeline %s", pipelineName)
+		return fmt.Errorf("Could not set pipeline for pipeline %s: %s", pipelineName, err)
+	}
+
+	err = data.Set("pipeline_id", p.Id)
+	if err != nil {
+		return fmt.Errorf("Could not set pipeline_id for pipeline %s: %s", pipelineName, err)
```

Comment:
Generally speaking, TF providers don't export the unique ID twice. The TF'y thing to do would be to just use `.id`, which is set by `.SetId()` below. (minor)

Reply:
The pipeline_id is needed as an output of the resource so that it can be referred to by e.g. "pipeline" stages in other pipelines.  I don't think HCL scripts can get at the "Id" set via SetId, correct?  So I think exporting it twice (once as the "pipeline_id" attribute, once as SetId()) is correct.

Reply:
`SetId()` is exposed as `${some_resource.foo.id}` automatically by TF.

Reply:
I will say there's precedent for this as some AWS resources use ARN for `SetId()`/`.id` _and_ export as `.arn`. It's minor to the point of being an FYI. I say :shipit:

Reply:
Ah, I was not aware of that!  That works for data sources as well?  If so, I'll submit a PR removing pipeline_id (breaking change, but I think the project is in a state where we can break consumers, @ethanfrogers ?)

Reply:
@joshuarowe-salesforce @joestump I _think_ it's okay if we make breaking changes right now. I still consider this project to be in an alpha state.

Reply:
@joshuarowe-salesforce yeah, works for both data and regular resources. 👍